### PR TITLE
fix: send max behavior

### PR DESCRIPTION
**Pull request by NeOMakinG**
## Description

- We were almost never using `sendMax` except if we were spamming it 2 times because the input validation was triggered, meaning that the sendMax value was set to false by it
- We were emptying error message in case it's been filled before the isAmountPositive effect, which was wrong as you could continue with a balance set as 0
- When sendMax, we were removing fee amount regardless if it's native token or not, fixed it by not removing them it's a native token only

## Issue (if applicable)

No issue yet

## Risk
High
> High Risk PRs Require 2 approvals

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

## Testing
- Test send max for all the chains, verify that fee estimation is right and amounts are right

### Engineering
n/a

### Operations
n/a
- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

## Screenshots (if applicable)
https://live.blockcypher.com/btc/tx/5c460fef9c003d862940d2453314ed1ed2f609527d44365e53954af33d806e53/
https://jam.dev/c/43cb58fa-40bc-4ef6-b5ca-713a14615390

https://jam.dev/c/fb6ef363-a215-4124-b686-da45be627685
https://basescan.org/tx/0x0be5c21b0a5b3bece6df65d07cfdfc0a6f299d82fca6f0690b8ae715d667eace

Cosmos, DOGECoin...
https://jam.dev/c/125c82f7-1d62-4d0c-9094-75b902224a63

### CODE REVIEW

**Review by woodenfurniture — APPROVED** [review]
# Testing notes

✅ Send ETH.ARB to self account `#1`, though it spun forever and required a page refresh to see the funds ⚠️ looks unrelated to this PR though.

https://jam.dev/c/1d891209-e431-4c7e-8015-f259a2fc68b5

**Comment by NeOMakinG** [comment]
> # Testing notes
> ✅ Send ETH.ARB to self account `#1`, though it spun forever and required a page refresh to see the funds ⚠️ looks unrelated to this PR though.
> 
> https://jam.dev/c/1d891209-e431-4c7e-8015-f259a2fc68b5
> 
> 

I think it would be worth to give it a bit more of tests if you don't mind: 
- UTXOs
- Cosmos SDK chains (thorchain/cosmos)
- Solana

I trust @gomesalexandre on the fact that this part is very touchy, specially on the `sendMax` logic

File: src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx [hunk 1]
```diff
@@ -275,11 +277,16 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
 
     const networkFee = bnOrZero(bn(fastFee).div(`1e${feeAsset.precision}`))
 
-    const maxCrypto = bnOrZero(cryptoHumanBalance).minus(networkFee)
+    const maxCrypto =
+      feeAsset.assetId !== assetId
+        ? bnOrZero(cryptoHumanBalance)
+        : bnOrZero(cryptoHumanBalance).minus(networkFee)
+
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx:284]: Yeah that one was most definitely definitely off, nice catch @NeOMakinG!

**Review by gomesalexandre — APPROVED** [review]
Tested locally, and spotted no regressions, great work ser 🐐 send max is definitely less flaky of a feature now!

### Dogecoin and USDC send max estimates

- Send max estimates are matching develop for DOGE ✅ 
- The wrong fee deduction is fee-deducted for USDC ✅ 
- Send maxes still go through and are actual max sends i.e all UTXOs reconciliated over a single one ✅ 

### BTC max sends

https://jam.dev/c/32f6b0be-76fb-469b-87ac-98c841422fd5
 
- Max sends estimates are still happy ✅  (yes, that max looks odd and extremely low but is correct, as with my UTXO set, it takes ~$35 USD of BTC to take all my inputs in and send to a single addy)

### ETH Max sends

https://jam.dev/c/25068a9b-e87c-4abe-8abb-ce56029155c7

- Send max estimates/sends are still happy ✅ 
- Non send-max estimates/sends are still happy ✅

**Pull request merged**